File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    let answer = '';
    let map = new Map();
    
    for (const p of participant) {
        if (map.has(p)) map.set(p, map.get(p) + 1)
        else map.set(p, 1)
    }
    
    for (const c of completion) {
        map.set(c, map.get(c) - 1)
    }
    
    const mapArr = map.entries()
    answer = [...mapArr].filter(([name, num]) => num !== 0)
    
    return answer[0][0];
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    dict = {}
    for i in participant:
        if i in dict:
            dict[i] += 1
        else:
            dict[i] = 1
    
    for c in completion:
        dict[c] -= 1
        
        
    for key in dict.keys():
        if dict[key] > 0:
            return key